openeo/extra/job_management.py

```diff
@@ -886,3 +936,191 @@ def create_job_db(path: Union[str, Path], df: pd.DataFrame, *, on_exists: str =
     else:
         raise NotImplementedError(f"Initialization of {type(job_db)} is not supported.")
     return job_db
+
+
+class UDPJobFactory:
+    """
+    Batch job factory based on a parameterized process definition
+    (e.g a user-defined process (UDP) or a remote process definition),
+    to be used together with :py:class:`MultiBackendJobManager`.
+
+    Usage example with a remote process definition:
+
+    .. code-block:: python
+
+        from openeo.extra.job_management import (
+            MultiBackendJobManager,
+            create_job_db,
+            UDPJobFactory,
+        )
+
+        # Job creator, based on a parameterized openEO process definition
+        job_starter = UDPJobFactory(
+            namespace="https://example.com/my_process.json",
+        )
+
+        # Initialize job database from dataframe, with parameters to use.
+        df = pd.DataFrame(...)
+        job_db = create_job_db("jobs.csv").initialize_from_df(df)
+
+        # Create and run job manager
+        job_manager = MultiBackendJobManager(...)
+        job_manager.run_jobs(job_db=job_db, start_job=job_starter)
+
+    The factory will take care of filling in the process parameters
+    based on matching column names in dataframe from the job database,
+    with some additional override/fallback options:
+
+    -   When provided, ``parameter_column_map`` will be consulted
+        for resolving a parameter name (key) to a desired column name (value).
+    -   One common case is handled automatically as convenience functionality.
+
+        When:
+
+        - ``parameter_column_map`` is not provided (or set to ``None``),
+        - and there is a *single parameter* that accepts inline GeoJSON geometries,
+        - and the dataframe is a GeoPandas dataframe with a *single geometry* column,
+
+        then this parameter and this geometries column will be linked automatically.
+
+    -   If a parameter can not be matched with a column by name as described above,
+        a default value will be picked,
+        first by looking in ``parameter_defaults`` (if provided),
+        and then by looking up the default value from the parameter schema in the process definition.
+    -   Finally if no (default) value can be determined and the parameter
+        is not flagged as optional, an error will be raised.
+
+    :param process_id: (optional) openEO process identifier.
+        Can be omitted when working with a remote process definition
+        given as URL in the ``namespace`` parameter.
+    :param namespace: (optional) openEO process namespace.
+        Typically used to provide a URL to a remote process definition.
+    :param parameter_defaults: Default values for process parameters,
```

Comment:
does each row then need to contain a list with all default values?

Reply:
no the `parameter_defaults` is an argument of the `UDPJobFactory` constructor